src/backend/pipeline/pipeline.py

```diff
@@ -108,9 +108,13 @@ async def run(self, imgs: Mat | list[Mat]) -> Data:
 
         # Run the chain
         iterator: Module | None = self.head
+        failed: bool = False
         while iterator is not None:
             # Run the module
-            data = iterator.run(data)
+            try:
+                data = iterator.run(data)
+            except Exception as exception:
+                failed = True
```

Comment:
I would catch this outside of the `while`? I wouldn't run the subsequent modules if the previous ones fail.

Reply:
it depends, you can save a lot for GOPS if you do